### DISCUSSION TITLE
[Run issue]:

### DISCUSSION THREAD

**Discussion by datboigbenga** [topas-nbio/TOPAS-nBio#37] [Run-Problem]
### Contact Details

eruolagbenga@gmail.com

### What was the issue?

When i try to install topas nBio, i get errors.
using the Cmake 
cmake -DTOPAS_EXTENSIONS_DIR=/Applications/ttopas_extensions/TOPAS-nBio
make -j8

### TOPAS-nBio category

Geometry

### Version

3.0

### What Operative System (OS) are you running on?

MacOS

### Relevant log output

```shell
/Applications/topas/extensions/DNACoordinates.cc:19:10: In file included from fatal error: /Applications/topas/extensions/ClusterAlgoSpatial.cc:In file included from 17/Applications/topas/extensions/ClusterAlgoBasePair.cc:17:
:
In file included from 'stdio.h' file not found/Applications/topas/extensions/AddChromosomeDomain.cc/Applications/topas/extensions/ClusterAlgoBasePair.hh/Applications/topas/extensions/ClusterAlgoSpatial.hh:
:In file included from 2319/Applications/topas/extensions/ClusterSBPoints.cc:::
:101925:::
In file included from  10/Applications/topas/extensions/ClusterSBPoints.hh: fatal error: /Applications/topas/extensions/AddChromosomeDomain.hh::2720:
:fatal error: 10:/Applications/topas/extensions/SBPoint.hh 'vector' file not found:fatal error: 27'vector' file not found
:
10'stdio.h' file not found: In file included from 
/Applications/topas/extensions/ClusteringAlgo.cc:fatal error: 25:
In file included from 'assert.h' file not found/Applications/topas/extensions/ClusteringAlgo.hh:
27:
In file included from /Applications/topas/extensions/ClusterSBPoints.hh:27:
/Applications/topas/extensions/SBPoint.hh:27:10: fatal error: 'assert.h' file not found
            19   19   271927 | 20 |  |  |  | ####i | #niiicn#iclinlucunndnldcuccleeud ll <udedu vee<d <ees<a vtsedccs<taeosirtstoe.rrro>>h.>h
t>.
      
| h            >|          ^~~~~~~~

|       
               ^~~~~~~~         ^~~~~~~~~~| | 

         ^~~~~~~~~~         ^~~~~~~~~

 <stdio.h>
      |          ^~~~~~~~~
In file included from /Applications/topas/extensions/ChromosomeParser.cc:9:
In file included from /Applications/topas/extensions/ChromosomeParser.hh:11:
In file included from /Applications/topas/Geant4Headers/globals.hh:35:
In file included from /Applications/topas/Geant4Headers/G4ios.hh:35:
/Applications/topas/Geant4Headers/G4Types.hh:74:10: fatal error: 'complex' file not found
   74 | #include <complex>
      |          ^~~~~~~~~
In file included from /Applications/topas/extensions/ChromosomeParameterisation.cc:24:
In file included from /Applications/topas/extensions/ChromosomeParameterisation.hh:27:
In file included from /Applications/topas/Geant4Headers/G4VPVParameterisation.hh:42:
/Applications/topas/Geant4Headers/G4Types.hh:74:10: fatal error: 'complex' file not found
   74 | #include <complex>
      |          ^~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/extensions.dir/extensions/ClusterSBPoints.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
1 error generated.
1 error generated.
make[2]: *** [CMakeFiles/extensions.dir/extensions/ChromosomeParser.cc.o] Error 1
1 error generated.
make[2]: *** [CMakeFiles/extensions.dir/extensions/ClusteringAlgo.cc.o] Error 1
1 error generated.
make[2]: *** [CMakeFiles/extensions.dir/extensions/ClusterAlgoSpatial.cc.o] Error 1
make[2]: *** [CMakeFiles/extensions.dir/extensions/ClusterAlgoBasePair.cc.o] Error 1
1 error generated.
make[2]: *** [CMakeFiles/extensions.dir/extensions/AddChromosomeDomain.cc.o] Error 1
1 error generated.
make[2]: *** [CMakeFiles/extensions.dir/extensions/DNACoordinates.cc.o] Error 1
1 error generated.
make[2]: *** [CMakeFiles/extensions.dir/extensions/ChromosomeParameterisation.cc.o] Error 1
make[1]: *** [CMakeFiles/extensions.dir/all] Error 2
make: *** [all] Error 2
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tmasilela** [comment]
Greetings,

Unless I am mistaken, it seems like you're trying to install the latest version of TOPAS-nBio with an older version of TOPAS. Since you're using MacOS, please refer to this [quickStart guide](https://github.com/OpenTOPAS/OpenTOPAS/blob/main/OpenTOPAS_quickStart_MacOS.md) on the official repository to first download the newest version of TOPAS (OpenTOPAS v4.0.0).

Secondly, please ensure that the path you are providing to the cmake command is the correct path, corresponding to the location of your TOPAS-nBio folder. In the command your provided there are two "t's" in the name of your extensions directory. Hope that helps, and good luck with the installation!

All the best,
Thongchai

**Reply by datboigbenga** [reply]
Thank you very much. I tried installing Open Topas , i am getting error when i try running

dyld[33518]: Library not loaded: @rpath/libQt5OpenGL.5.dylib
  Referenced from: <024B023A-685D-33EB-A31B-459617E81516> /Applications/TOPAS/OpenTOPAS-install/bin/topas
  Reason: tried: '/Applications/GEANT4/geant4-install/lib/libQt5OpenGL.5.dylib' (no such file), '/Applications/TOPAS/OpenTOPAS-install/lib/libQt5OpenGL.5.dylib' (no such file), '/libQt5OpenGL.5.dylib' (no such file), '/Applications/TOPAS/OpenTOPAS-install/Frameworks/libQt5OpenGL.5.dylib' (no such file), '/Applications/TOPAS/OpenTOPAS-install/Frameworks/libQt5OpenGL.5.dylib' (no such file)
/Users/apple/shellScripts/topas: line 7: 33518 Abort trap: 6           /Applications/TOPAS/OpenTOPAS-install/bin/topas $1
